classes/User.php

```diff
@@ -588,14 +588,14 @@ public static function verifyEmailLinkCode($code, User $user = null) {
 			$query = 'UPDATE u_users
 						SET email_verified = 1,
 							email_verification_code = null,
-							email_verification_code_time = null
+							email_verification_code_time = NOW()
 						WHERE email_verification_code = ?
 							AND email_verification_code_time > DATE_SUB(NOW(), INTERVAL ? DAY)';
 		} else {
 			$query = 'UPDATE u_users
 						SET email_verified = 1,
 							email_verification_code = null,
-							email_verification_code_time = null
+							email_verification_code_time = NOW()
```

Comment:
This is incorrect. This field must be null when code is not set. Instead the code needs to be explicitly set to NOW() when user is created elsewhere in the code where it relies on INSERT without value to be setting current time.